src/main/java/Duke.java

```diff
@@ -1,10 +1,588 @@
+import java.io.*;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+import CustomException.DukeException;
+
+// These are not Strings; they are constant variables
+/* enum Op {
+    TODO,
+    DEADLINE,
+    EVENT,
+    LIST,
+    DONE,
+    DELETE,
+    BYE
+} */
+
+/**
+ * Represents a chatbot to store list of things to do. A <code>Duke</code> object corresponds to
+ * a chatbot that stores a list of things to do.
+ */
 public class Duke {
-    public static void main(String[] args) {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException | IOException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public static void main(String[] args) throws IOException {
+        Duke duke = new Duke("./src/main/java/database.txt");
+        duke.run();
+    }
+
+    public void run() throws IOException {
```

Comment:
I think you can use logo directly without instantiating it since you only use it once.